pkg/syncer/admissionhandler/cnscsi_admissionhandler.go

```diff
@@ -119,10 +119,6 @@ func (h *CSISupervisorWebhook) Handle(ctx context.Context, req admission.Request
 			admissionResp := validatePVC(ctx, &req.AdmissionRequest)
 			resp.AdmissionResponse = *admissionResp.DeepCopy()
 		}
-	} else if req.Kind.Kind == "VolumeSnapshot" {
-		if featureGateBlockVolumeSnapshotEnabled {
-			resp = validateSnapshotOperationSupervisorRequest(ctx, req)
-		}
```

Comment:
This is dangerous, so ideally, we need to allow the creation if the snapshot quota fss is enabled, you would need to wait for @nikhilbarge change to go in https://github.com/kubernetes-sigs/vsphere-csi-driver/pull/2783. The logic would be something like:
If storage quota fss is enabled- allow snapshot creation on the supervisor, else, validate it to see if it's a create and reject

Reply:
updated to check FSS SnapshotQuotaM2 and allow snapshot create request if FSS enabled